### PR TITLE
Add manifests for registry

### PR DESCRIPTION
**Pull request by zoetrope**

### CODE REVIEW

File: test/registry_test.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+}
+
+func testRegistry() {
+	It("should cache containers on mirror registries", func() {
+		type catalog struct {
+			Repositories []string `json:"repositories"`
+		}
+
+		Eventually(func() error {
```
Comment (masa213f) [inline test/registry_test.go:89]: This Eventually is too long.
How about checking one by one?

File: test/registry_test.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const podsYaml = `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: elastic-elasticsearch
+  namespace: registry
```
Comment (masa213f) [inline test/registry_test.go:19]: I think it is better to create a namespace for testing.

**Review by masa213f — CHANGES_REQUESTED** [review]

File: argocd-config/overlays/osaka0/registry-quay.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+kind: Application
+metadata:
+  name: registry-quay
+  namespace: argocd
+spec:
+  source:
+    helm:
+      parameters:
+        - name: "proxy.secretName"
+          value: quay-credential-t5c62bctdb
```
[the remark this answers is not in the document]
Comment (zoetrope) [reply argocd-config/overlays/osaka0/registry-quay.yaml:11]: What is a volume file?

File: docs/maintenance.md [hunk 4]
```diff
@@ -239,6 +239,10 @@ Download the upstream tar ball from [releases](https://github.com/topolvm/pvc-au
 $ kustomize build ./config/default > /path/to/pvc-autoresizer/base/upstream.yaml
 ```
 
+## registry
+
+Check [the release notes](https://github.com/docker/distribution/releases).
```
[the remark this answers is not in the document]
Comment (zoetrope) [reply docs/maintenance.md:244]: The base does not exist.
This is completely original.

File: registry/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,103 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: registry-{{ .Values.instance }}
+  namespace: registry
+  labels:
+    app.kubernetes.io/name: registry
+    app.kubernetes.io/instance: {{ .Values.instance }}
```
[the remark this answers is not in the document]
Comment (zoetrope) [reply registry/templates/deployment.yaml:8]: What do you recommend?

File: test/registry_test.go [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+spec:
+  containers:
+  - command:
+    - /usr/local/bin/pause
```
[the remark this answers is not in the document]
Comment (zoetrope) [reply test/registry_test.go:23]: If the pod is working fine, does it work?
I want to pull container images again.

File: docs/maintenance.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (masa213f) [reply docs/maintenance.md:244]: So please write a note. We can't judge the manifest is scratch or not.

File: test/registry_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (masa213f) [reply test/registry_test.go:22]: Sorry, it does not work...
I tried the following manifest. It did not pull the image when the container restarting.

```
apiVersion: v1
kind: Pod
metadata:
  name: pull-test
  namespace: sandbox
spec:
  containers:
  - command:
    - sleep
    - "10"
    image: quay.io/cybozu/ubuntu-debug:18.04
    imagePullPolicy: Always
    name: ubuntu
  securityContext:
    runAsGroup: 1000
    runAsUser: 1000
```

File: docs/maintenance.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (zoetrope) [reply docs/maintenance.md:244]: 🙆‍♂️

**Review by masa213f — APPROVED** [review]
Thank you!

**Pull request merged**